### DISCUSSION TITLE
Image panel colormode/colormap

### DISCUSSION THREAD

**Discussion by Wambomango** [foxglove/community#1034] [Q&A] [answered]
Hello there!
I was wondering how to access the colormode and colormap features hinted on in  the [docs](https://docs.foxglove.dev/docs/visualization/panels/image/).
I assume, you should be able to colorize monochrome images, yet I cannot see these fields in the settings.
Thank you for your help!

**Comment by defunctzombie** [comment, marked as answer]
<img width="479" alt="image" src="https://github.com/user-attachments/assets/198f9477-762d-4457-af40-fd74961c2a24">

Are you using one of these two formats for your image data as indicated in the docs?

**Reply by jtbandes** [reply]
Internal tracking issue: FG-4406

**Reply by Wambomango** [reply]
So, I got it to work with your hint (how did I miss this am I stupid!?). Though I have still two open questions:
-On the one hand, when using a CompressedImage instead of an Image, the colormap option is not available, even when using 16U1 as a base format. Is this intentional?
-On the other hand, is it planned to support colormaps also for the other single channel formats? This should be fairly easy to implement unless I am being stupid right now (probably)

My ideal scenario would be to compress 8bit/pixel images, which after publishing get uncompressed and colorized for visualization within foxglove.
With the current constraints, I would have to publish 16bit/pixel uncompressed images.
This adds a lot of overhead, such that I might as well publish the colorized/compressed images and have a lower data-rate.

Thanks again for your help!

**Reply by defunctzombie** [reply]
Which compression are you thinking for 8bit/pixel images? Like a PNG?

**Reply by Wambomango** [reply]
> -On the one hand, when using a CompressedImage instead of an Image, the colormap option is not available, even when using 16U1 as a base format. Is this intentional?

If what you are refering to is this, then I used JPEG compression in my test.

Correct me if I am wrong, but ultimately, shouldnt the colormap feature be able to work for any compression?
After decompression, the image is available just like it would be having published the non-compressed image in the first place.
Therefore, the colorization algorithm can remain the same, just that a compressed image needs to be decompressed before colorization which is necessary anyways.

I hope I am not missing your point.
